### ISSUE TITLE
ifnb@assays$SCT$data, ifnb@assays$RNA$data

### ISSUE THREAD

**Issue by zznx** [SingleR-inc/SingleR#272]
Hi,  for the SignleR function, which data should I choose as input

ifnb@assays$SCT$data
ifnb@assays$RNA$data

```
Seurat::SCTransform(
  object,
  assay = "RNA",
  new.assay.name = "SCT",
  reference.SCT.model = NULL,
  do.correct.umi = TRUE,
  ncells = 5000,
  residual.features = NULL,
  variable.features.n = 3000,
  variable.features.rv.th = 1.3,
  vars.to.regress = NULL,
  do.scale = FALSE,
  do.center = TRUE,
  clip.range = c(-sqrt(x = ncol(x = object[[assay]])/30), sqrt(x = ncol(x =
    object[[assay]])/30)),
  vst.flavor = "v2",
  conserve.memory = FALSE,
  return.only.var.genes = TRUE,
  seed.use = 1448145,
  verbose = TRUE,
  ...
)



do.correct.umi           	
Place corrected UMI matrix in assay counts slot; default is TRUE
```

**Comment by LTLA** [comment]
See comments in #98.

**tl;dr** it probably doesn't matter but just keep things simple and use the raw counts.

**Comment by zznx** [comment]
```
logNormCounts(ifnb@assays$RNA$counts) 
or
DefaultAssay(ifnb) = 'RNA'
as.SingleCellExperiment(ifnb) 
```
That would be the right choice, I have also seen the source code before, thank you very much for your reply

**Comment by LTLA** [comment]
Don't really know what you're asking, but monotonic transformations like `logNormCounts()` on the test matrix won't affect the results; **SingleR** is all based on rankings within each cell.

**Comment by zznx** [comment]
I've got your answer, thank you for your time
I learned about SingleR's algorithm a long time ago
Thank you anyway!
